backend/service-user/src/main/java/g55/cs3219/backend/userService/config/SecurityConfiguration.java

```diff
@@ -34,22 +34,22 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .csrf(csrf -> csrf.disable())
                 .authorizeHttpRequests(authorize -> authorize
                         // Allow registration and login for all users
-                        .requestMatchers("/auth/**", "/users").permitAll()
+                        .requestMatchers("/api/auth/**", "/api/users").permitAll()
 
                         // Get all users - restricted to admin only
-                        .requestMatchers(HttpMethod.GET, "/users").hasRole("ADMIN")
+                        .requestMatchers(HttpMethod.GET, "/api/users").hasRole("ADMIN")
 
                         // Get user by ID - admins can access any, non-admins can only access their own (handled in service)
-                        .requestMatchers(HttpMethod.GET, "/users/{userId}").authenticated()
+                        .requestMatchers(HttpMethod.GET, "/api/users/{userId}").authenticated()
 
                         // Update user - admins can update any, non-admins can only update their own (handled in service)
-                        .requestMatchers(HttpMethod.PUT, "/users/{userId}").authenticated()
+                        .requestMatchers(HttpMethod.PUT, "/api/users/{userId}").authenticated()
```

Comment:
Great job on making this change! Maybe next time we can separate this to a separate PR first, easier to review and make changes on FE side as well